plan/plan.go

```diff
@@ -40,6 +41,10 @@ type Plan interface {
 	// If index can be used, a new index plan is returned, 'filtered' is true.
 	// If no index can be used, the original plan is returned and 'filtered' return false.
 	Filter(ctx context.Context, expr expression.Expression) (p Plan, filtered bool, err error)
+
+	// Next returns the next row of data and rowKeys, nil data means there is no more data to return.
+	// Aggregation plan will fetch all the data at the first call.
+	Next(ctx context.Context) (data []interface{}, rowKeys []*RowKeyEntry, err error)
```

Comment:
I think we should define a Row struct as the return type, e.g, if we have to add another xxx, we can change in Row

```
type Row struct {
    Data []interface{}
    Keys []*RowKeyEntry
    // other 
}
```